lib/trycourier/brands.rb

```diff
@@ -0,0 +1,105 @@
+require "trycourier/version"
+require "net/http"
+require "json"
+require "openssl"
+
+module Courier
+  # generic error to catch all exceptions
+  class CourierAPIException < StandardError; end
+
+  class Brands
+    @@key = "/brands"
+
+    def initialize(base_url, session)
+      @base_url = base_url + @@key
+      @session = session
+    end
+
+    def checkErr(res)
+      code = res.code.to_i
+      obj = JSON.parse res.read_body
+
+      if code == 200
+        obj
+      elsif (message = obj["Message"].nil? ? obj["message"] : obj["Message"])
+        err = "#{code}: #{message}"
+        raise CourierAPIException, err
+      end
+    end
+
+    def checkErr2(res)
+      code = res.code.to_i
+      if code >= 400
+        obj = JSON.parse res.read_body
+        if (message = obj["Message"].nil? ? obj["message"] : obj["Message"])
+          err = "#{code}: #{message}"
+          raise CourierAPIException, err
+        end
+      end
+      res
+    end
```

Comment:
It feels like this logic shouldn't be duplicated in each endpoint class. Maybe we actually need to build out the `CourierAPIExpection` class instead of defining it as generic in each file.